### ISSUE TITLE
Field extractor crashes if extracted directive does not exist

### ISSUE THREAD

**Issue by gabrielhase** [upfrontIO/livingdocs-engine#188]
When the design contains a field extractor entry for a directive that does not exist then it crashes. This is an example with a typo:
```
"matches": ["head.txet"]
```
Of course, this should read `head.text` so the designer made an error when entering the match. Nevertheless, the engine should not crash (as it does now) if it comes by a typo. I would propose that upon seeing a directive name that does not exist the engine should print an error message.

**Comment by DaRaFF** [comment]
@gabrielhase  When i create a new component f.e. `livingdoc.createComponent('advertisement')` and the component doesn't exist, the engine also crashes. Should i also open an issue for that or is this behaviour wanted?

**Comment by gabrielhase** [comment]
Crashes are always bad :smile: It shouldn't do any intelligent in that case but at least throw a controlled error message.

**Comment by DaRaFF** [comment]
It does throw an intelligent error messsage, but it's not possible to continue with rendering the channel then. In my view it would be better to just ignore this operation and continue, yes?

**Comment by gabrielhase** [comment]
Hmm.. no in that case I wouldn't agree. We shouldn't render documents with invalid markup. @peyerluk what do you think?

**Comment by DaRaFF** [comment]
As a consequence, we have to handle the error in render decorator. I already did that, here is an example -> https://github.com/nzzdev/livingdocs-render-decorator/blob/master/src/middleware/pre-barrier-adder.coffee#L4-L8. But as an external consumer of the engine, do we really need to handle errors like that?

**Comment by gabrielhase** [comment]
I guess in our understanding the way to handle these errors is to correct the design so that there are no components that do not exist. It's just pretty dangerous to silently continue on this kind of stuff because it leads to data-loss, i.e., the journalist has some kind of data in his edited document that never gets published (because it is silently skipped). IMO we should prevent publishing altogether and send back an error message to the user.

**Comment by DaRaFF** [comment]
This approach only works, when we can render every document in their appropriate version. In the current situation (without migrations and only beeing able to index all documents with the same design version), this is error prone.

Our approach is: better show 95% of an publication then no publication.

**Comment by gabrielhase** [comment]
Yes, I agree that it is error prone at the moment and we need to get control back over this. I would just be careful with doing "hacks" now that lead to ever more loss of control. @peyerluk will hopefully show something from his migration/design/metadata/everything concept on Monday and then we can make a plan how we move to this target step by step.
